errors.go

```diff
@@ -0,0 +1,81 @@
+package notmuch
+
+// Copyright © 2015 The go.bindings Authors. Authors can be found in the AUTHORS file.
+// Licensed under the GPLv3 or later.
+// See COPYING at the root of the repository for details.
+
+// #cgo LDFLAGS: -lnotmuch
+// #include <stdlib.h>
+// #include <notmuch.h>
+import "C"
+import "errors"
+
+type status C.notmuch_status_t
+
+var (
+	// ErrOutOfMemory is returned when an Out of memory occured.
+	ErrOutOfMemory = statusErr(C.NOTMUCH_STATUS_OUT_OF_MEMORY)
```

Comment:
I kept this one exposed and `checkOOM` now panic with it so we can compare it to the error received in recover.